mne_connectivity/wsmi.py

```
@@ -0,0 +1,483 @@
+    except Exception as e:
+        logger.error(f"Error during symbolic transformation: {e}")
+        raise
+
+    n_unique_symbols = count.shape[1]
+    if (
+        sym.shape[0] != n_channels_picked
+        or sym.shape[2] != n_epochs
+        or count.shape[0] != n_channels_picked
+        or count.shape[2] != n_epochs
+    ):
+        raise ValueError(
+            f"""Symbolic transformation output has unexpected shape.
+            Got sym: {sym.shape}, count: {count.shape}.
+            Expected channels: {n_channels_picked}, epochs: {n_epochs}."""
+        )
+
+    wts = _get_weights_matrix(n_unique_symbols)
+
+    # --- 4. wSMI/SMI Computation (Jitted) ---
+    method_name = "wSMI" if weighted else "SMI"
+    logger.info(
+        f"""Computing {method_name} for {n_unique_symbols} unique symbols
+        (Numba-jitted)..."""
+    )
+    result = _wsmi_python_jitted(sym, count, wts, weighted)
+    # result is (n_channels_picked, n_channels_picked, n_epochs)
+
+    result_epoched = result.transpose(2, 0, 1)
+
+    # --- Packaging results ---
+    if n_channels_picked > 1:
+        # Extract upper triangular part (excluding diagonal) for connectivity object
+        triu_inds = np.triu_indices(n_channels_picked, k=1)
+        indices_list = list(zip(triu_inds[0], triu_inds[1]))
+
+        # Extract connectivity data for upper triangular connections only
+        result_conn_data = np.zeros((n_epochs, len(indices_list), 1))
+        for epoch_idx in range(n_epochs):
+            for conn_idx, (i, j) in enumerate(indices_list):
+                result_conn_data[epoch_idx, conn_idx, 0] = result_epoched[
+                    epoch_idx, i, j
+                ]
+    else:
+        # For single channel or no channels, create empty connectivity
+        logger.info(
+            f"Only 1 channel selected, {method_name} connectivity will be empty."
+        )
+        result_conn_data = np.empty((n_epochs, 0, 1))
+        indices_list = []
```

I would expect this to raise an error rather than returning an empty connectivity object. Perhaps also with an `indices` parameter problematic cases like this could be caught early?

---

Done! Please resolve conversation, or let me know if anything’s unclear.